Content.Server/RoleTimers/RoleTimerManagers.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Content.Server.Database;
+using Robust.Shared.Network;
+
+namespace Content.Server.RoleTimers
+{
+    public sealed class RoleTimerManager
+    {
+        [Dependency] private readonly IServerDbManager _db = default!;
+
+        private RoleTimer GetRoleTimer(NetUserId userId, string role)
+        {
+            return _db.GetRoleTimer(userId.UserId, role).Result;
```

Comment:
Do not do `.Result` under any circumstance. Pass the `Task` through and make the calling code async.